### PR TITLE
fix for #7602, and address feedback

### LINKED ISSUE #7602: All galleries are loaded from the cloud on localhost.
Try loading the editor in localhost and see the URL of galleries. It's downloading from makecode.com.

image

Due to https://github.com/microsoft/pxt/commit/ed6180765c67f822b312eb3ea27f6e4b8faf4a3b#commitcomment-44040161

### PR DESCRIPTION
**Pull request by eanders-MS**
Reverted changes in:
* cloud.ts
* util.ts
* fileworkspace.ts

Moved `BACKEND` selection to auth.ts. This gives backend devs flexibility to run against localhost or not.

fixes #7602

### CODE REVIEW

File: webapp/src/auth.ts [hunk 1]
```diff
@@ -343,14 +349,23 @@ type ApiResult<T> = {
     errmsg: string;
 };
 
+const BACKEND_DEFAULT = "";
+const BACKEND_PROD = "https://www.makecode.com";
```
Comment (pelikhan) [inline webapp/src/auth.ts:353]: to be friendly to OSS, a 3rd party should be able to override this. Our API used ``/api/`` to allow this.

Comment (pelikhan) [reply webapp/src/auth.ts:353]: (something to ponder about)

Comment (eanders-MS) [reply webapp/src/auth.ts:353]: In the prod environment the url is relative and will hit whatever backend the site was served from. This is a dev-only override.

Comment (eanders-MS) [reply webapp/src/auth.ts]: I scoped it to only apply when running on localhost/dev.
https://github.com/microsoft/pxt/pull/7604/files/ca9311a75f5f0aa58f911043a42544b2708ae7c3..e5bea850a769873bda22234bc26f4c967a27fc34

**Review by pelikhan — APPROVED** [review]

**Pull request merged**